artifactory/services/utils/resultutils.go

```diff
@@ -53,13 +53,20 @@ func (ad *ArtifactDetails) ToBuildInfoArtifact() buildinfo.Artifact {
 	artifact := buildinfo.Artifact{Checksum: buildinfo.Checksum{}}
 	artifact.Sha1 = ad.Checksums.Sha1
 	artifact.Md5 = ad.Checksums.Md5
+	artifact.Sha256 = ad.Checksums.Sha256
 	// Artifact name in build info as the name in artifactory
 	filename, _ := fileutils.GetFileAndDirFromPath(ad.ArtifactoryPath)
 	artifact.Name = filename
 	if i := strings.LastIndex(filename, "."); i != -1 {
 		artifact.Type = filename[i+1:]
 	}
-	artifact.Path = ad.ArtifactoryPath
+	// Remove the repository from Artifact's path.
+	// Since a build can be promoted, the build-info may be incorrect (builds are immutable).
+	if i := strings.Index(ad.ArtifactoryPath, "/"); i != -1 {
+		artifact.Path = ad.ArtifactoryPath[i+1:]
+	} else {
```

Comment:
Is there a known case where the path at this stage does not include the repository? If there is, let's explain it in a comment here. If there isn't, let's not use a "defensive" approach, and return an error with a clear "unexpected" message inside this else.